RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.cc

```diff
@@ -78,298 +94,301 @@ void TSGForOI::produce(edm::StreamID sid, edm::Event& iEvent, const edm::EventSe
                                                                 measurementTrackerH->geometricSearchTracker()->negTecLayers();
-  edm::ESHandle<TrackerTopology> tTopo_handle;
-  iSetup.get<TrackerTopologyRcd>().get(tTopo_handle);
-  const TrackerTopology* tTopo = tTopo_handle.product();
+  //edm::ESHandle<TrackerTopology> tTopo_handle;
+  //iSetup.get<TrackerTopologyRcd>().get(tTopo_handle);
+  //const TrackerTopology* tTopo = tTopo_handle.product();
 
-  //	Get the suitable propagators: 
+  // Get suitable propagators
   std::unique_ptr<Propagator> propagatorAlong = SetPropagationDirection(*propagatorAlongH,alongMomentum);
   std::unique_ptr<Propagator> propagatorOpposite = SetPropagationDirection(*propagatorOppositeH,oppositeToMomentum);
 
-  edm::ESHandle<Propagator> SmartOpposite;
+  // Stepping Helix Propagator for propogation from uon system to tracker
   edm::ESHandle<Propagator> SHPOpposite;
-  iSetup.get<TrackingComponentsRecord>().get("hltESPSmartPropagatorAnyOpposite", SmartOpposite);
   iSetup.get<TrackingComponentsRecord>().get("hltESPSteppingHelixPropagatorOpposite", SHPOpposite);
 
-  //	Loop over the L2's and make seeds for all of them:
+  // Loop over the L2's and make seeds for all of them
   LogTrace(theCategory) << "TSGForOI::produce: Number of L2's: " << l2TrackCol->size();
-  for (unsigned int l2TrackColIndex(0);l2TrackColIndex!=l2TrackCol->size();++l2TrackColIndex){
+  for (unsigned int l2TrackColIndex(0); l2TrackColIndex!=l2TrackCol->size(); ++l2TrackColIndex) {
+
     const reco::TrackRef l2(l2TrackCol, l2TrackColIndex);
-    std::unique_ptr<std::vector<TrajectorySeed> > out(new std::vector<TrajectorySeed>());
+
+    // Container of Seeds
+    std::vector<TrajectorySeed> out;
     LogTrace("TSGForOI") << "TSGForOI::produce: L2 muon pT, eta, phi --> " << l2->pt() << " , " << l2->eta() << " , " << l2->phi() << endl;
-    
+ 
     FreeTrajectoryState fts = trajectoryStateTransform::initialFreeState(*l2, magfieldH.product());
+
     dummyPlane->move(fts.position() - dummyPlane->position());
     TrajectoryStateOnSurface tsosAtIP = TrajectoryStateOnSurface(fts, *dummyPlane);
     LogTrace("TSGForOI") << "TSGForOI::produce: Created TSOSatIP: " << tsosAtIP << std::endl;
     
-    // get the TSOS on the innermost layer of the L2. 
+    // Get the TSOS on the innermost layer of the L2
     TrajectoryStateOnSurface tsosAtMuonSystem = trajectoryStateTransform::innerStateOnSurface(*l2, *geometryH, magfieldH.product());
     LogTrace("TSGForOI") << "TSGForOI::produce: Created TSOSatMuonSystem: " << tsosAtMuonSystem <<endl;
     
     LogTrace("TSGForOI") << "TSGForOI::produce: Check the error of the L2 parameter and use hit seeds if big errors" << endl;
+
     StateOnTrackerBound fromInside(propagatorAlong.get());
     TrajectoryStateOnSurface outerTkStateInside = fromInside(fts);
-      
-    StateOnTrackerBound fromOutside(&*SmartOpposite);
+
+    StateOnTrackerBound fromOutside(&*SHPOpposite);
```

Comment:
I thought that the smart propagator was supposed to give essentially equivalent result (using SHP outside the tracker volume and [I'm not sure, was it RK] inside the tracker volume).
Is this change simply technical or did you find problems with the smart propagator?
IIRC, it's used also in some other muon reco (global and standalone)

Reply:
Yes, We have just chosen the stepping helix. Basically it is the same. Outside the tracker volume stepping helix is widely used and inside , it is propgagatorWithMaterial.  We have also seen this combination gives better performance.